management/connection_test.go

```diff
@@ -44,6 +99,44 @@ func TestConnection(t *testing.T) {
 		t.Logf("%v\n", c)
 	})
 
+	t.Run("Update custom scripts and configuration values", func(t *testing.T) {
+		id := c.ID
+		c.ID = ""       // read-only
+		c.Name = ""     // read-only
+		c.Strategy = "" // read-only
+		c.Options.CustomScripts = map[string]interface{}{
+			"get_user": "function(email, callback) { return callback(null) }",
+		}
+		c.Options.Configuration = map[string]interface{}{
+			"KEY_1": "VALUE_1",
+			"KEY_2": "VALUE_2",
+		}
+
+		err = m.Connection.Update(id, c)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if c.Options.CustomScripts["get_user"] != "function(email, callback) { return callback(null) }" {
```

Comment:
The js function makes this a bit hard to read, perhaps we can figure out a cleaner way to make this comparison? I was thinking this whole test can be shrinked to something like this:

```
id := c.ID
c.ID = ""       // read-only
c.Name = ""     // read-only
c.Strategy = "" // read-only
c.Options = ConnectionOptions{
  CustomScripts: map[string]interface{}{"get_user": "function(email, callback) { return callback(null) }"},
  Configuration: map[string]interface{}{"foo": "bar"},
}

cc := c // make a copy so we can compare later

err = m.Connection.Update(id, c)
if err != nil {
  t.Error(err)
}

if c.Options.CustomScripts["get_user"] != cc.Options.CustomScripts["get_user"] {
  t.Fatal(`unexpected result for "get_user" custom script`)
}

if _, exist := c.Options.Configuration["foo"]; !exist {
  t.Fatal(`missing key "foo"`)
}
```